ethereumj-core/src/main/java/org/ethereum/vm/DataWord.java

```diff
@@ -81,18 +86,26 @@ public DataWord(byte[] data) {
             this.data = ByteUtil.EMPTY_BYTE_ARRAY;
         else if (data.length == 32)
             this.data = data;
-        else if (data.length <= 32)
-            System.arraycopy(data, 0, this.data, 32 - data.length, data.length);
-        else
-            throw new RuntimeException("Data word can't exceed 32 bytes: " + data);
+        else if (data.length <= 32) {
+            byte[] bytes = ZERO.getData();
```

Comment:
To preserve mutability there should be a brand new empty array allocated and then filled with given data. The same thing is with above case

Reply:
But doing it that way we won't be able to efficiently use this constructor in inner `DataWord` allocations where `DataWord` itself cares about immutability. I see two options (i) make this constructor private and use for inner purposes, for outer create a method `DataWord.from(byte[] data)` with all required checks and copies (ii) do it vise versa

Reply:
Yep, this looks pretty odd. I see that `getData()` makes a copy, but `new byte[32]` suites better here